Formula/c/certifi.rb

```diff
@@ -15,24 +15,35 @@ class Certifi < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:  "1f1fc985a1c89bd40c73b17e3dfbf5483cb0417c8d4d12e2be66158a503ab169"
   end
 
-  depends_on "python@3.11" => [:build, :test]
-  depends_on "python@3.12" => [:build, :test]
   depends_on "python@3.13" => [:build, :test]
+  depends_on "python@3.11" => :test # keep on oldest Python to support
   depends_on "ca-certificates"
 
   def pythons
-    deps.map(&:to_formula).sort_by(&:version).filter { |f| f.name.start_with?("python@") }
+    deps.filter_map { |dep| dep.to_formula if dep.name.start_with?("python@") }
+        .sort_by(&:version)
   end
 
   def install
-    pythons.each do |python|
-      python_exe = python.opt_libexec/"bin/python"
-      system python_exe, "-m", "pip", "install", *std_pip_args(build_isolation: true), "."
+    # Ensure uniform bottles
+    inreplace "README.rst", "/usr/local", HOMEBREW_PREFIX
+
+    python = deps.find { |dep| dep.name.start_with?("python@") && dep.build? }
+                 .to_formula
+    python_exe = python.opt_libexec/"bin/python"
+    system python_exe, "-m", "pip", "install", *std_pip_args(build_isolation: true), "."
+
+    # Use brewed ca-certificates PEM file instead of the bundled copy
+    site_packages = prefix/Language::Python.site_packages(python_exe)
+    rm(site_packages/"certifi/cacert.pem")
+    (site_packages/"certifi").install_symlink Formula["ca-certificates"].pkgetc/"cert.pem" => "cacert.pem"
 
-      # Use brewed ca-certificates PEM file instead of the bundled copy
-      site_packages = Language::Python.site_packages("python#{python.version.major_minor}")
-      rm prefix/site_packages/"certifi/cacert.pem"
-      (prefix/site_packages/"certifi").install_symlink Formula["ca-certificates"].pkgetc/"cert.pem" => "cacert.pem"
+    # Create symlinks so that the Python bindings can be used with alternative Python versions
+    oldest_python_version = Language::Python.major_minor_version(pythons.first.opt_libexec/"bin/python")
+    other_pythons = python.versioned_formulae.filter { |f| f.version >= oldest_python_version }
+    other_pythons.each do |other_python|
+      other_python_version = Language::Python.major_minor_version(other_python.opt_libexec/"bin/python")
```

Comment:
`major_minor_version` here actually tries to run the `python3` of the specified version and then reads the output for the version.

You would think that shouldn't work when the wanted Python interpreter isn't installed...

Reply:
Probably will need to get version from `Formula#version` instead or just hard code the minimum in a separate method, e.g. `def oldest_python_version`. If we don't care about testing oldest Python, then latter would simplify dependency tree.

Would be simpler to install library in all Pythons, but I recall multiple user reports of `brew link` failures when we introduced `certifi`. I expect we will see more complaints if we backport to older Pythons.

Reply:
Well it actually gives the correct version because the error message produced when it tries to run the binary still contains the version we want. So it works, but I guess is technically incorrect. Should probably fix that in `brew`.

Reply:
> So it works, but I guess is technically incorrect. Should probably fix that in `brew`.

Yeah. Not exactly the behavior we expect.

---

Luckily in this part of handling, we already have the formula so can easily switch to that:
```suggestion
      other_python_version = other_python.version.major_minor
```